File: backend/src/main/java/org/example/backend/model/entities/Invoice.java
package org.example.backend.model.entities;


import lombok.*;
import org.apache.pdfbox.util.filetypedetector.FileType;
import org.springframework.data.annotation.Id;

import java.time.LocalDate;

/**
 * Abrechnungen
 */
@With
@Builder
public record Invoice(
        @Id
        String id,
        String docNumber,//Belegnummer
        LocalDate date,
        double amount,//Preis
        String purpose,//Verwendungszweck
        boolean isInvoice,//Rechnung oder Zahlungsbeleg
        String fileName,//Dateiname
        String fileType,//Dateityp
        byte[] file//File in base64
) {
}

File: frontend/src/model/Invoice.ts
export type Invoice = {
    id: string;
    docNumber: string; // Belegnummer
    date: string; // kommt als dd.MM.yyyy vom Backend
    amount: number; // Preis
    purpose: string; // Verwendungszweck
    isInvoice: boolean; // Rechnung oder Zahlungsbeleg
    fileName: string; // Dateiname
    fileBase64: string | null; // Base64 String vom Backend
};
